WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -13,15 +13,54 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
         self.cardReaderConnectionViewModel = cardReaderConnectionViewModel
     }
 
-    // Creates a unique `CartProduct` from a `Product`, and adds it to the Cart
     func addProductToCart(_ product: POSProduct) {
-        let cartProduct = CartProduct(id: UUID(), product: product, quantity: 1)
-        productsInCart.append(cartProduct)
+        if product.stockQuantity > 0 {
+            reduceInventory(product)
+
+            let cartProduct = CartProduct(id: UUID(), product: product, quantity: 1)
+            productsInCart.append(cartProduct)
+        } else {
+            // TODO: Handle out of stock
+            // wp.me/p91TBi-bcW#comment-12123
+            return
+        }
+    }
+
+    func reduceInventory(_ product: POSProduct) {
+        guard let index = products.firstIndex(where: { $0.itemID == product.itemID }) else {
+            return
+        }
+        let updatedQuantity = product.stockQuantity - 1
+        let updatedProduct = POSProduct(itemID: product.itemID,
+                                    productID: product.productID,
+                                    name: product.name,
+                                    price: product.price,
+                                    stockQuantity: updatedQuantity)
+        products[index] = updatedProduct
+    }
+
+    func restoreInventory(_ product: POSProduct) {
```

Comment:
This could simplified into an unique method for reducing/increasing, no strong reason to keep them separate at this point, just preference.